uniffi_bindgen/src/bindings/swift/gen_swift.rs

```diff
@@ -139,66 +155,196 @@ impl<'config, 'ci> ModuleMap<'config, 'ci> {
-        Self { config, ci }
+    pub fn members(&self) -> Vec<Box<dyn CodeDeclaration + 'a>> {
+        let ci = self.ci;
+        vec![
+            Box::new(object::SwiftObjectRuntime::new(ci)) as Box<dyn CodeDeclaration>,
+            Box::new(callback_interface::SwiftCallbackInterfaceRuntime::new(ci))
+                as Box<dyn CodeDeclaration>,
+        ]
+        .into_iter()
+        .chain(
+            ci.iter_enum_definitions().into_iter().map(|inner| {
+                Box::new(enum_::SwiftEnum::new(inner, ci)) as Box<dyn CodeDeclaration>
+            }),
+        )
+        .chain(ci.iter_function_definitions().into_iter().map(|inner| {
+            Box::new(function::SwiftFunction::new(inner, ci)) as Box<dyn CodeDeclaration>
+        }))
+        .chain(
+            ci.iter_object_definitions().into_iter().map(|inner| {
+                Box::new(object::SwiftObject::new(inner, ci)) as Box<dyn CodeDeclaration>
+            }),
+        )
+        .chain(
+            ci.iter_record_definitions().into_iter().map(|inner| {
+                Box::new(record::SwiftRecord::new(inner, ci)) as Box<dyn CodeDeclaration>
+            }),
+        )
+        .chain(
+            ci.iter_error_definitions().into_iter().map(|inner| {
+                Box::new(error::SwiftError::new(inner, ci)) as Box<dyn CodeDeclaration>
+            }),
+        )
+        .chain(
+            ci.iter_callback_interface_definitions()
+                .into_iter()
+                .map(|inner| {
+                    Box::new(callback_interface::SwiftCallbackInterface::new(inner, ci))
+                        as Box<dyn CodeDeclaration>
+                }),
+        )
+        .collect()
+    }
+
+    pub fn initialization_code(&self) -> Vec<String> {
+        let oracle = &self.oracle;
+        self.members()
+            .into_iter()
+            .filter_map(|member| member.initialization_code(oracle))
+            .collect()
+    }
+
+    pub fn declaration_code(&self) -> Vec<String> {
+        let oracle = &self.oracle;
+        self.members()
+            .into_iter()
+            .filter_map(|member| member.definition_code(oracle))
+            .chain(
+                self.ci
+                    .iter_types()
+                    .into_iter()
+                    .filter_map(|type_| oracle.find(&type_).helper_code(oracle)),
+            )
+            .collect()
+    }
+
+    pub fn imports(&self) -> Vec<String> {
+        let oracle = &self.oracle;
+        let mut imports: Vec<String> = self
+            .members()
+            .into_iter()
+            .filter_map(|member| member.import_code(oracle))
+            .flatten()
+            .chain(
+                self.ci
+                    .iter_types()
+                    .into_iter()
+                    .filter_map(|type_| oracle.find(&type_).import_code(oracle))
+                    .flatten(),
+            )
+            .collect::<HashSet<String>>()
```

Comment:
super nit: any reason we first collect into a set and then into the Vec?

Reply:
This is one of the best features!  If multiple components require the same import, we'll only generate 1 import statement.

Super Super nit: Maybe collect into `BTreeSet` to de-dupe and sort at the same type.